BraveShared/Analytics/DAU.swift

```diff
@@ -199,85 +221,63 @@ public class DAU {
         
-        log.debug("Dau stat params, daily: \(daily), weekly: \(weekly), monthly:\(monthly), lastPingDate: \(lastPingDate)")
-        if !daily && !weekly && !monthly {
-            // No changes, no ping
+        // No changes, no ping
+        if pings.isEmpty {
             return nil
         }
         
         return dauParams(daily, weekly, monthly)
     }
 }
 
-extension DateComponents {
-    /// Returns date of current week's monday in YYYY-MM-DD format
-    var weeksMonday: String {
-        var isSunday: Bool {
-            guard let weekday = weekday else {
-                log.error("Weekday is nil")
-                return false
-            }
-            return weekday == 1
-        }
-        
-        // Make sure all required date components are set.
-        guard let _ = day, let _ = month, let _ = year, let weekday = weekday else {
-            log.error("Date components are missing")
-            return ""
-        }
-        
-        guard let today = DAU.calendar.date(from: self) else {
-            log.error("Cannot create date from date components")
-            return ""
-        }
-        
-        let dayInSeconds = 60 * 60 * 24
-        // Sunday is first weekday so we need to handle this day differently, can't just substract it.
-        let sundayToMondayDayDifference = 6
-        let dayDifference = isSunday ? sundayToMondayDayDifference : weekday - 2 // -2 because monday is second weekday
-        
-        let monday = Date(timeInterval: -TimeInterval(dayDifference * dayInSeconds), since: today)
-        let mondayComponents = DAU.calendar.components([.day, .month, .year], from: monday)
+extension Date {
+    /// Returns date of current week's monday in YYYY-MM-DD formatted String
+    var mondayOfCurrentWeekFormatted: String? {
+        // We look for a previous monday because Sunday is considered a beggining of a new week using default gregorian calendar.
+        // For example if today is Sunday, the next Monday using Calendar would be the day after Sunday which is wrong.
+        // That's why backward search may sound counter intuitive.
+        guard let monday = self.next(.monday, direction: .backward, considerToday: true) else { return nil }
+        
+        let dateFormatter = DateFormatter()
+        dateFormatter.dateFormat = "yyyy-MM-dd"
+        
+        return dateFormatter.string(from: monday)
+    }
+    
+    func next(_ weekday: Weekday, direction: Calendar.SearchDirection = .forward, considerToday: Bool = false) -> Date? {
+        let calendar = DAU.calendar
+        let components = DateComponents(weekday: weekday.rawValue)
         
-        guard let mYear = mondayComponents.year, let mMonth = mondayComponents.month, let mDay = mondayComponents.day else {
-            log.error("First monday of the week components are nil")
-            return ""
+        if considerToday && calendar.component(.weekday, from: self) == weekday.rawValue {
```

Comment:
Considering this is an extension on `Date` which can be any date, couldn't `calendar.component(.weekday, from: self)` be any Monday and not be today?

Reply:
I'm not sure if I understand you.
This can be any weekday, depending on what date we are going to pass and against which weekday we are comparing.
Should I make it more specific and hide behind `private` accessor?

Reply:
So yeah I don't think based on usage this is a problem because its always called on `today`, was just a bit confusing to understand when not paying attention to the call site/only paying attention to the fact that this is a method on any `Date`

Reply:
I think renaming to `considerSelf` would make sense